ApplicationCode/GrpcInterface/RiaGrpcCaseService.cpp

```diff
@@ -364,6 +522,145 @@ grpc::Status RiaGrpcCaseService::GetCellInfoForActiveCells( grpc::ServerContext*
+///
+//--------------------------------------------------------------------------------------------------
+grpc::Status RiaGrpcCaseService::GetCellCenterForActiveCells( grpc::ServerContext*           context,
+                                                              const rips::CellInfoRequest*   request,
+                                                              rips::CellCenters*             reply,
+                                                              RiaActiveCellInfoStateHandler* stateHandler )
+{
+    return stateHandler->assignCellCentersReply( reply );
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+grpc::Status RiaGrpcCaseService::GetCellCornersForActiveCells( grpc::ServerContext*           context,
+                                                               const rips::CellInfoRequest*   request,
+                                                               rips::CellCornersArray*        reply,
+                                                               RiaActiveCellInfoStateHandler* stateHandler )
+{
+    return stateHandler->assignCellCornersReply( reply );
+}
+
+RiaSelectedCellsStateHandler::RiaSelectedCellsStateHandler()
+    : m_request( nullptr )
+    , m_eclipseCase( nullptr )
+    , m_currentItem( 0u )
+{
+}
+
+Status RiaSelectedCellsStateHandler::init( const rips::CaseRequest* request )
+{
+    CAF_ASSERT( request );
+    m_request = request;
+
+    RimCase* rimCase = RiaGrpcServiceInterface::findCase( m_request->id() );
+    m_eclipseCase    = dynamic_cast<RimEclipseCase*>( rimCase );
+
+    if ( !m_eclipseCase )
+    {
+        return grpc::Status( grpc::NOT_FOUND, "Eclipse Case not found" );
+    }
+
+    if ( !m_eclipseCase->eclipseCaseData() || !m_eclipseCase->eclipseCaseData()->mainGrid() )
+    {
+        return grpc::Status( grpc::NOT_FOUND, "Eclipse Case Data not found" );
+    }
+
+    return grpc::Status::OK;
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+Status RiaSelectedCellsStateHandler::assignNextSelectedCell( rips::SelectedCell*                          cell,
+                                                             const std::vector<RiuEclipseSelectionItem*>& items )
+{
+    while ( m_currentItem < items.size() )
+    {
+        size_t itemToTry = m_currentItem++;
+
+        const RiuEclipseSelectionItem* item = items[itemToTry];
+        CVF_ASSERT( item->type() == RiuSelectionItem::ECLIPSE_SELECTION_OBJECT );
+        assignSelectedCell( cell, item );
+        return grpc::Status::OK;
+    }
+    return Status( grpc::OUT_OF_RANGE, "We've reached the end. This is not an error but means transmission is finished" );
+}
+
```

Comment:
Missing some separator comments on methods in this file.

Reply:
Fixed.